### ISSUE TITLE
Translation BLEU scores

### ISSUE THREAD

**Issue by EmreOzkose** [pr0mila/MediBeng-Whisper-Tiny#1]
Hi @pr0mila,

Great work. Thanks for sharing the code. Have you been able to check translation BLEU scores?

**Comment by pr0mila** [comment]
Hi @EmreOzkose,

Thanks for reaching out! For now, you can integrate the BLEU score using the following adjustments:

1. Load the BLEU metric from the `evaluate` library using `evaluate.load("bleu")`.
2. Modify the `compute_metrics` function to calculate and return the BLEU score instead of WER.

In the next version, I will include BLEU score evaluation as part of the update. Feel free to let me know if you need any more details!

Here’s a code snippet for the changes to update the `compute_metrics` function for bleu score:

```python

# Initialize the evaluation metric for BLEU
metric_bleu = evaluate.load("bleu")

def compute_metrics(pred):
    pred_ids = pred.predictions
    label_ids = pred.label_ids

    # Replace -100 with the pad_token_id (to ensure ignored tokens are not counted)
    label_ids[label_ids == -100] = tokenizer.pad_token_id

    # Convert predictions to the correct format
    if isinstance(pred_ids, tuple):
        pred_ids = pred_ids[0]
    
    # Convert to numpy array if needed
    if isinstance(pred_ids, list):
        pred_ids = np.array(pred_ids)
    if isinstance(label_ids, list):
        label_ids = np.array(label_ids)

    # Ensure we have the correct shape for predictions
    if len(pred_ids.shape) == 3:  # (batch_size, num_beams, sequence_length)
        pred_ids = pred_ids[:, 0, :]  # Take the first beam if using beam search
    elif len(pred_ids.shape) == 2:  # (batch_size, sequence_length)
        pass
    else:
        raise ValueError(f"Unexpected prediction shape: {pred_ids.shape}")

    # Convert to list of lists for the tokenizer
    pred_ids = pred_ids.tolist()
    label_ids = label_ids.tolist()

    # Decode predictions and labels using tokenizer
    pred_str = tokenizer.batch_decode(pred_ids, skip_special_tokens=True)
    label_str = tokenizer.batch_decode(label_ids, skip_special_tokens=True)

    # Ensure both are correctly aligned for BLEU calculation
    if len(pred_str) != len(label_str):
        raise ValueError(f"Mismatch between number of predictions and references: {len(pred_str)} vs {len(label_str)}")

    # Calculate BLEU score
    bleu_score = metric_bleu.compute(predictions=pred_str, references=[[ref] for ref in label_str])

    # Optionally log the BLEU score
    logging.info(f"BLEU Score: {bleu_score['bleu']}")

    return {"eval_bleu": bleu_score["bleu"]}


```
then Update the in the Seq2SeqTrainingArguments, change the metric_for_best_model to `bleu` instead of `wer`

```python
training_args = Seq2SeqTrainingArguments(
    output_dir=OUTPUT_DIR,
    per_device_train_batch_size=TRAIN_BATCH_SIZE,
    gradient_accumulation_steps=GRADIENT_ACCUMULATION_STEPS,
    learning_rate=LEARNING_RATE,
    warmup_steps=WARMUP_STEPS,
    max_steps=MAX_STEPS,
    gradient_checkpointing=True,
    fp16=False,
    eval_strategy="steps",
    per_device_eval_batch_size=EVAL_BATCH_SIZE,
    save_steps=SAVE_STEPS,
    eval_steps=EVAL_STEPS,
    logging_steps=LOGGING_STEPS,
    report_to=["tensorboard"],
    load_best_model_at_end=True,
    metric_for_best_model="bleu",
    greater_is_better=False,
    push_to_hub=False,
    predict_with_generate=True,
    generation_max_length=GENERATION_MAX_LENGTH,
    logging_dir=f"{OUTPUT_DIR}/runs/{current_time}",  # Add timestamp to TensorBoard logs
)

```

**Comment by EmreOzkose** [comment]
Great! Thank you for sharing the modified version of compute metrics. I want to train this on English-Turkish data when I am available. When it is done, I can share my observations and results. They may help other people.
